Wrappers/Python/ccpi/viewer/CILViewerBase.py

```diff
@@ -0,0 +1,278 @@
+        self.helpActor.GetPositionCoordinate().SetCoordinateSystemToNormalizedDisplay()
+        self.helpActor.GetPositionCoordinate().SetValue(0.1, 0.5)
+        self.helpActor.VisibilityOff()
+
+        # axis orientation widget
+        om = vtk.vtkAxesActor()
+        ori = vtk.vtkOrientationMarkerWidget()
+        ori.SetOutlineColor(0.9300, 0.5700, 0.1300)
+        ori.SetInteractor(self.iren)
+        ori.SetOrientationMarker(om)
+        ori.SetViewport(0.0, 0.0, 0.4, 0.4)
+        ori.SetEnabled(1)
+        ori.InteractiveOff()
+        self.orientation_marker = ori
+
+        # holder for list of actors
+        self.actors = {}
+
+        #initial Window/Level
+        self.InitialLevel = 0
+        self.InitialWindow = 0
+
+        #ViewerEvent
+        self.event = ViewerEventManager()
+
+        self.histogramPlotActor = vtk.vtkXYPlotActor()
+        self.histogramPlotActor.ExchangeAxesOff()
+        self.histogramPlotActor.SetXLabelFormat("%g")
+        self.histogramPlotActor.SetXLabelFormat("%g")
+        self.histogramPlotActor.SetAdjustXLabels(3)
+        self.histogramPlotActor.SetXTitle("Level")
+        self.histogramPlotActor.SetYTitle("N")
+        self.histogramPlotActor.SetXValuesToValue()
+        self.histogramPlotActor.SetPlotColor(0, (0, 1, 1))
+
+    def getInteractor(self):
+        return self.iren
+
+    def getRenderer(self):
+        return self.ren
+
+    def getRenderWindow(self):
+        '''returns the render window'''
+        return self.renWin
+
+    def getSliceOrientation(self):
+        return self.sliceOrientation
+
+    def setActiveSlice(self, sliceno):
+        self.slicenos[self.getSliceOrientation()] = sliceno
+
+    def getActiveSlice(self):
+        return self.slicenos[self.getSliceOrientation()]
+
+    def getSliceColorWindow(self):
+        return self.imageSlice.GetProperty().GetColorWindow()
+
+    def getSliceColorLevel(self):
+        return self.imageSlice.GetProperty().GetColorLevel()
+
+    def getSliceWindowLevelFromRange(self, cmin, cmax):
+        # set the level to the average between the percentiles
+        level = (cmin + cmax) / 2
+        # accommodates all values between the level an the percentiles
+        window = cmax - cmin
+
+        return window, level
+
+    def startRenderLoop(self):
+        self.iren.Start()
+
+    def getImageHistogramStatistics(self, method):
+        '''
+        returns histogram statistics for either the image
+        or gradient of the image depending on the method
+        '''
+        ia = vtk.vtkImageHistogramStatistics()
+        if method == 'scalar':
+            ia.SetInputData(self.img3D)
+        else:
+            grad = vtk.vtkImageGradientMagnitude()
+            grad.SetInputData(self.img3D)
+            grad.SetDimensionality(3)
+            grad.Update()
+            ia.SetInputData(grad.GetOutput())
+        ia.Update()
+        return ia
+
+    def getVolumeMapRange(self, percentiles, method):
```

Comment:
I'm not a fan of the term `volume` but as long as it's documented that with `slice` we mean `a 2D slice of the 3D image` and with `volume` we mean `whole 3D image`, I'm fine with that

Reply:
What would you prefer instead of 'volume' to mean the whole image?

Reply:
It's clearly confusing: VTK calls it image, and with volume, I often think of the volume render. That's why I'd stick to image for the whole dataset and slice for the 2D subset.

Reply:
I changed to 'image'